ocp_deployment/templates/compass-backend-template.yaml

```diff
@@ -528,6 +529,9 @@ parameters:
     - name: PUBLIC_KEY
       description: Personal public Key for encryption functions
       required: true
+    - name: CERTIFICATE
+      description: The certificate which is used for the encryption of the responses on the app (optional)
+      value: ''
```

Comment:
Can we unify the way to specify optional parameters? SSH_PRIVATE_KEY is also optional but there we don't set an empty value. It would be better if both variables are either initialized with an empty string as value (as now for CERTIFICATE) or to expand the documentation in line 109 on how to set the parameter if it is not required.

Reply:
The hint optional would not be required here (to make the description shorter), as "required" is not set to true, so this would be visible when listing the template parameters via the CLI.

Reply:
> Can we unify the way to specify optional parameters? SSH_PRIVATE_KEY is also optional but there we don't set an empty value. It would be better if both variables are either initialized with an empty string as value (as now for CERTIFICATE) or to expand the documentation in line 109 on how to set the parameter if it is not required.

SSH_KEY is set as "required: true" in the yaml, therefore it is not optional as per the template.

I personally would prefer to be able to completely omit optional parameters.

Reply:
Right, the remplate requires it but if the git repository is public it down't have to be set. 
I would however also include it. Then we should also set CERTIFICATE to **required: true**.

Reply:
But if CERTIFICATE is **required:true** we can't allow an empty string as that would override the default in the config (which is set to 'false') but an empty string is not a valid certificate and this case isn't checked for in the frontend/app therefore this would ultimately lead to issues.

To be fair: as far as i know, there is no check for the certificate (or the public/private keys) to be valid, so it's up to the user to check that, but that's another issue.